python/src/internal.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Internal APIs for Lance format.
+//!
+//! Python API binding that exposes various
+
+use lance::dataset::transaction::Transaction;
+use lance::io::commit::read_transaction_file;
+use lance::io::ObjectStore;
+use lance_core::format::Manifest;
+use lance_core::io::reader::read_manifest;
+use pyo3::prelude::*;
+
+use crate::RT;
+
+#[pyclass(name = "Manifest", module = "lance.internal")]
+#[derive(Debug)]
+pub struct PyManifest {
+    manifest: Manifest,
+}
+
+#[pymethods]
+impl PyManifest {
+    #[new]
+    fn new(path: String) -> PyResult<Self> {
+        RT.runtime.block_on(async {
+            // the store created is always at the root of the file system
+            let (store, _) = ObjectStore::from_uri(".").await.map_err(|e| {
```

Comment:
Does this actually work? I think this needs to be tested. I would think this would create a store with it's root at the current directory, not the root of the filesystem...